File: Week_01/G20200343040015/move-zeroes.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
    let k = 0;
    for(let i = 0; i<nums.length; i++) {
        if(nums[i] !== 0) {
            nums[k] = nums[i];
            if( i !== k ) {
                nums[i] = 0;
            };
            k += 1;
        }
    }
};

File: Week_01/G20200343040105/MoveZeroes.cpp
//283.移动0   

//双指针替换法
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int i = 0, j = 0;
        for (; i < nums.size(); ++i) {
            if (nums[i] != 0) {
                if (i != j) swap (nums[i], nums[++j]);
            }
            else ++j;
        }   
    }
};
//时间复杂度 O(n) 空间复杂度O(1)
